Sources/StreamChat/Utils/Codable+Extensions.swift

```diff
@@ -139,12 +139,16 @@ extension DateFormatter {
             return gmtDateFormatter.string(from: date)
         }
 
-        private static let gmtDateFormatter: DateFormatter = {
+        private static var locale: Locale = Locale(identifier: "en_US_POSIX")
+
+        private static var timeZone: TimeZone? = TimeZone(secondsFromGMT: 0)
+
+        private static var gmtDateFormatter: DateFormatter {
             let formatter = DateFormatter()
-            formatter.locale = Locale(identifier: "en_US_POSIX")
-            formatter.timeZone = TimeZone(secondsFromGMT: 0)
+            formatter.locale = locale
+            formatter.timeZone = timeZone
             return formatter
-        }()
+        }
```

Comment:
This will create a new DateFormatter every time. It is recommended by Apple that we don't do this since it is expensive to create a DateFormatter, especially if we are creating it every cell in a table view for example. 

The https://github.com/GetStream/stream-chat-swift/pull/1486 solution seems pretty solid to me